network/src/gossip.rs

```diff
@@ -318,6 +310,7 @@ impl<F, P> ChainContext for (F, P) where
 pub fn register_validator<C: ChainContext + 'static>(
```

Comment:
This function was always a bit awkward; I'd guess that we'd want to refactor things so that the network crate isn't responsible for instantiating all of these gossip validators. That would fix your "always `Some` in practice, `None` in tests" issue from below.

Reply:
Not a blocker for this PR, but probably how we want to do things in the future